mergin/client.py

```diff
@@ -700,41 +718,67 @@ def project_info(self, project_path_or_id, since=None, version=None):
             resp = self.get("/v1/project/{}".format(project_path_or_id), params)
         return json.load(resp)
 
-    def project_versions(self, project_path, since=None, to=None):
+    def project_versions_page(self, project_path, page, per_page=100, descending=False):
+        params = {"page": page, "per_page": per_page, "descending": descending}
+        resp = self.get("/v1/project/versions/paginated/{}".format(project_path), params)
+        resp_json = json.load(resp)
+        return resp_json["versions"]
+
+    def project_versions_count(self, project_path):
+        """
+        return the total count of versions
+
+        To note the requested informations are kept to the minimal
+        """
+        params = {"page": 1, "per_page": 1, "descending": False}
+        resp = self.get("/v1/project/versions/paginated/{}".format(project_path), params)
+        resp_json = json.load(resp)
+        return resp_json["count"]
+
+    def project_versions_in_range(self, project_path, since=None, to=None):
         """
         Get records of project's versions (history) in ascending order.
         If neither 'since' nor 'to' is specified it will return all versions.
 
         :param project_path: Project's full name (<namespace>/<name>)
-        :type project_path: String
+        :type project_path: String | Int
         :param since: Version to track project history from
-        :type since: String
+        :type since: String | Int
         :param to: Version to track project history to
         :type to: String
 
         :rtype: List[Dict]
         """
         versions = []
         per_page = 100  # server limit
-        num_since = int_version(since) if since else 1
-        num_to = int_version(to) if to else None  # we may not know yet
+
+        if type(since) == str:
+            num_since = int_version(since)
+        elif since == None:
+            num_since = 1
+        else:
+            # keep the since parameter as is
+            num_since = since
+
+        if type(to) == str:
+            num_to = int_version(to)
+        else:
+            # keep the to parameter as is
+            num_to = to
+
         start_page = math.ceil(num_since / per_page)
         if not num_to:
             # let's get first page and count
-            params = {"page": start_page, "per_page": per_page, "descending": False}
-            resp = self.get("/v1/project/versions/paginated/{}".format(project_path), params)
-            resp_json = json.load(resp)
-            versions = resp_json["versions"]
-            num_to = resp_json["count"]
+            versions = self.project_versions_page(project_path, start_page, per_page)
+            num_to = self.project_versions_count(project_path)
```

Comment:
I think that ```self.project_versions_count``` and ```self.project_versions_page``` are calling similar endpoint twice. This is not optimal. You can check count from project_versions_page directly.

Reply:
Indeed they call the same end point but the trade off is that we don't have to fetch & jsonload & return value twice in the function 

I can add count as the return value from `self.project_versions_page` but `self.project_versions_count` is really lightweight and only fetch 1 item instead of 100

Reply:
I think, use one call for that If it's possible . You can return items and count as result of project_versions_page. Then you can use this method directly in removed ```project_versions``` and also in this section

Reply:
Okay I changed it by return both the versions at this page and the total count using someting like that: 
`return resp_json["versions"], resp_json["count"]`